linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala

```diff
@@ -1311,13 +1443,18 @@ private[emitter] class FunctionEmitter(sjsGen: SJSGen) {
           allowSideEffects && test(lhs) && test(rhs)
 
         // Expressions preserving pureness (modulo NPE)
-        case Block(trees)            => trees forall test
-        case If(cond, thenp, elsep)  => test(cond) && test(thenp) && test(elsep)
-        case BinaryOp(_, lhs, rhs)   => test(lhs) && test(rhs)
-        case RecordSelect(record, _) => test(record)
-        case IsInstanceOf(expr, _)   => test(expr)
+        case Block(trees) =>
+          testAll(trees)
+        case If(cond, thenp, elsep) =>
+          test(cond) && test(thenp) && test(elsep) && !isRTLongType(tree.tpe)
```

Comment:
Couldn't we push down `allowUnsplittableLongs` in the if branches here? (I realize that doing this might unnecessarily complicate other parts of the codegen, but maybe it's worth a comment?)